src/Evolution/Particles/MonteCarlo/Scattering.cpp

```diff
@@ -55,22 +57,25 @@ DiffusionMonteCarloParameters::DiffusionMonteCarloParameters()
     const double TargetRho = sqrt(3. * cTau / 4.);
     // Find A such that F[TargetRho]=A
     // i.e. ScatteringRofP[A]=TargetRho
-    size_t cF = 0;
-    while (cF < p_pts && gsl::at(ScatteringRofP, cF) < TargetRho) {
-      cF++;
-    }
-    if (cF == p_pts) {
+    const auto upper_bracket = std::lower_bound(
+        ScatteringRofP.begin(), ScatteringRofP.end(), TargetRho);
+    if (upper_bracket == ScatteringRofP.end()) {
```

Comment:
is there any chance that this would find the first element of `ScatteringROfP` in which case there is no previous element?  If so there needs to be an ERROR or something handling that case

Reply:
I don't think so; TargetRho is at least 
sqrt(3*MinimumOpacityForDiffusion/4)=sqrt(9/4)
and 
ScatteringRofP[0] = 0
so it will return the second element or higher.

Reply:
I'll add a comment.